### PR TITLE
Config changes

### PR DESCRIPTION
**Pull request by RubenRBS**
As a PR mainly so I can have another pair of eyes to recheck the config changes.

This PR:
- Sorts the C & C++ libs and tools alphabetically
- Adds missing semver settings for some C++ Compilers
- Adds missing RSIC-V Clang versions
- Fixes how RISC-V Clang is configured. It's now less of a mess
- Adds missing Zig C & C++ compilers, along with Tincy changes

As you can see, I totally forgot to change branches before commiting them, so I had to branch off and now they are here. Sorry about that

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -336,272 +339,188 @@ compiler.clang_embed.semver=(std::embed)
...
-# An alias of the original name for this compiler needs to be maintained to
-# allow old URLs to continue to work
+# An alias of the original name for this compiler needs to be maintained to allow old URLs to continue to work
 compiler.armv8-full-clang-trunk.alias=armv8.5-clang-trunk
 
+################################
 # Clang for RISC-V
-group.rvclang.compilers=rv32-clang:rv32-clang1101:rv32-clang1100:rv32-clang1001:rv32-clang1000:rv32-clang901:rv32-clang900:rv64-clang:rv64-clang1101:rv64-clang1100:rv64-clang1001:rv64-clang1000:rv64-clang901:rv64-clang900
+group.rvclang.compilers=&rv32clang:&rv32clang
 group.rvclang.groupName=RISC-V Clang
```
Comment (RubenRBS) [inline etc/config/c++.amazon.properties:435]: I'm not 100% sure that this works for subgroups. Do both of them end up in the same optgroup in the compiler selector?

Comment (partouf) [reply etc/config/c++.amazon.properties:435]: is this also not meant to say rv64clang on the second one?

That said - I don't think there's much support for subgroups in general. So I would be hesitant to try it out

Comment (RubenRBS) [reply etc/config/c++.amazon.properties:435]: And this is exactly why it's a PR, I suck at spotting my sloppy c&p. Thank you!

File: etc/config/c++.amazon.properties [hunk 2]
```diff
@@ -336,272 +339,188 @@ compiler.clang_embed.semver=(std::embed)
...
 compiler.armv8-full-clang-trunk.options=-target aarch64-linux-gnu --gcc-toolchain=/opt/compiler-explorer/arm64/gcc-8.2.0/aarch64-unknown-linux-gnu --sysroot=/opt/compiler-explorer/arm64/gcc-8.2.0/aarch64-unknown-linux-gnu/aarch64-unknown-linux-gnu/sysroot -march=armv8.6-a+crypto+rcpc+sha3+sm4+profile+rng+memtag+sve2+sve2-bitperm+sve2-sm4+sve2-aes+sve2-sha3+tme
-# An alias of the original name for this compiler needs to be maintained to
-# allow old URLs to continue to work
+# An alias of the original name for this compiler needs to be maintained to allow old URLs to continue to work
 compiler.armv8-full-clang-trunk.alias=armv8.5-clang-trunk
 
+################################
 # Clang for RISC-V
-group.rvclang.compilers=rv32-clang:rv32-clang1101:rv32-clang1100:rv32-clang1001:rv32-clang1000:rv32-clang901:rv32-clang900:rv64-clang:rv64-clang1101:rv64-clang1100:rv64-clang1001:rv64-clang1000:rv64-clang901:rv64-clang900
+group.rvclang.compilers=&rv32clang:&rv32clang
```
Comment (RubenRBS) [inline etc/config/c++.amazon.properties:434]:
```suggestion
group.rvclang.compilers=&rv32clang:&rv64clang
```

Comment (RubenRBS) [reply etc/config/c++.amazon.properties:434]: As per @partouf comment

File: etc/config/c++.amazon.properties [hunk 3]
```diff
@@ -926,7 +843,7 @@ group.avr.compilers=avrg454:avrg464:avrg540:avrg920:avrg930:avrg1030:avrg1100
 group.avr.groupName=AVR GCC
 group.avr.baseName=AVR gcc
 group.avr.isSemVer=true
-group.avr.supportsBinary=true
```
Comment (RubenRBS) [inline etc/config/c++.amazon.properties:929]: The parent group already sets this to true

File: etc/config/c++.amazon.properties (hunk 1 above)
Comment (RubenRBS) [reply etc/config/c++.amazon.properties:435]: imagen
Argh

File: etc/config/c++.amazon.properties [hunk 4]
```diff
@@ -336,272 +339,189 @@ compiler.clang_embed.semver=(std::embed)
...
-# allow old URLs to continue to work
+# An alias of the original name for this compiler needs to be maintained to allow old URLs to continue to work
 compiler.armv8-full-clang-trunk.alias=armv8.5-clang-trunk
 
+################################
 # Clang for RISC-V
-group.rvclang.compilers=rv32-clang:rv32-clang1101:rv32-clang1100:rv32-clang1001:rv32-clang1000:rv32-clang901:rv32-clang900:rv64-clang:rv64-clang1101:rv64-clang1100:rv64-clang1001:rv64-clang1000:rv64-clang901:rv64-clang900
-group.rvclang.groupName=RISC-V Clang
-group.rvclang.supportsBinary=false
+group.rvclang.compilers=&rv32clang:&rv64clang
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties:434]: thanks

File: etc/config/c++.amazon.properties [hunk 5]
```diff
@@ -802,44 +726,36 @@ group.gcc64arm.compilers=aarchg54:arm64g630:arm64g640:arm64g730:arm64g820:arm64g
 group.gcc64arm.isSemVer=true
 group.gcc64arm.objdumper=/opt/compiler-explorer/arm64/gcc-10.2.0/aarch64-unknown-linux-gnu/aarch64-unknown-linux-gnu/bin/objdump
 group.gcc64arm.instructionSet=aarch64
+group.gcc64arm.baseName=ARM64 gcc
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties:729]: So much better!

File: etc/config/c++.amazon.properties (hunk 3 above)
Comment (mattgodbolt) [reply etc/config/c++.amazon.properties:929]: :+1:

File: lib/compilers/tinyc.js [hunk 6]
```diff
@@ -36,7 +36,7 @@ export class TinyCCompiler extends BaseCompiler {
             if (!_.some(userOptions, (opt) => opt === '-E')) {
                 filters.binary = true;
             }
-            return ['-o', this.filename(outputFilename)];
+            return ['-g', '-o', this.filename(outputFilename)];
```
Comment (mattgodbolt) [inline lib/compilers/tinyc.js:39]: Should maybe be a separate change (or did you say that in the PR?)

**Review by mattgodbolt — APPROVED** [review]
Everything I looked at looked OK. But my eyes started to go a bit funny looking at such a big diff.... We really need to do something more proactive about this. Mayyyyybe it's time to autogenerate these files (maybe still checking in the output? Jinja template? ugh)

File: lib/compilers/tinyc.js (hunk 6 above)
Comment (RubenRBS) [reply lib/compilers/tinyc.js:39]: Yeah, I screwed my git-fu and had to push a few unreleated changes as part of the pr, sorry :(

**Comment by RubenRBS** [comment]
I'll bring staging up with this once more and check if the api response for the list of C++ compilers differs from prod, and if everything looks ok, I'll merge the branch

**Pull request merged**